custom_components/smartthinq_sensors/wideq/devices/washerDryer.py

```diff
@@ -708,6 +751,71 @@ async def select_start_course(self, course_name: str) -> None:
+        course_id = self._get_course_infos().get(self._selected_course)
+        n_course_key = self.get_course_key(CourseType.COURSE)
+        course_info = self._get_course_details(n_course_key, course_id)
+        if not course_info:
+            raise ValueError("Course info not available")
+        # _LOGGER.debug("select_start_course, course_info: %s", course_info)
+
+        self._course_overrides.clear()
+        self._course_overrides_lists.clear()
+        for func_key in course_info["function"]:
+            value = func_key.get("value")
+            default = func_key.get("default")
+            selectable = func_key.get("selectable")
+
+            if selectable is None:
+                continue
+
+            _LOGGER.debug("select_start_course(%s), set overrides for %s - default: %s, selectable: %s", course_name, value, default, selectable)
+            self._course_overrides[value] = default
+            self._course_overrides_lists[value] = selectable
+
+
+    def _select_enabled(self, select_name: str) -> bool:
+        """Return if specified select is enabled."""
+        enabled = self.select_course_enabled and self._selected_course and self._course_overrides_lists.get(select_name)
+        if (not enabled) and (select_name in self._course_overrides):
+            del self._course_overrides[select_name]
+        return enabled
+
+    def _select_start_option(self, option: str, option_friendly_name: str, option_selected: str) -> None:
+        """Select a secific option for remote start."""
+        permitted_options = self._course_overrides_lists.get(option)
+        if permitted_options and option_selected in permitted_options:
+            self._course_overrides[option] = option_selected
+        else:
+            raise ServiceValidationError(f"{option_selected} is invalid and will be ignored. {option_friendly_name} must be one of {permitted_options}")
+
+    @property
+    def select_temp_enabled(self) -> bool:
+        """Return if select temp is enabled."""
+        return self._select_enabled('temp')
+        
+    async def select_start_temp(self, temp_name: str) -> None:
+        """Select a secific water temperature for remote start."""
+        self._select_start_option('temp', 'Water Temp', temp_name)
+
+    @property
+    def select_rinse_enabled(self) -> bool:
+        """Return if select rinse is enabled."""
+        return self._select_enabled('rinse')
+
+    async def select_start_rinse(self, rinse_name: str) -> None:
+        """Select a secific rinse option for remote start."""
+        self._select_start_option('rinse', 'Rinse Option', rinse_name)
+
+    @property
+    def select_spin_enabled(self) -> bool:
+        """Return if select spin is enabled."""
+        return self._select_enabled('spin')
+        
+    async def select_start_spin(self, spin_name: str) -> None:
+        """Select a secific spin for remote start."""
+        self._select_start_option('spin', 'Spin Speed', spin_name)
+
```

Comment:
This will make all the logic very complex, because there are different combination that can vary depending on selected course. I would prefer to just add some parameter to the service call and allow to change this value calling the HA service with prober parameters, that will be ignored if not valid.

Reply:
@ollo69 , 

> This will make all the logic very complex, because there are different combination that can vary depending on selected course

You are correct the ability to override a temp, spin or rinse depends on the WM model and the course selected.  The PR handles this by extracting the necessary data for course info and storing it in `_course_overrides_lists`.  The code should not required updating for different or future VM models unless a new setting is introduced (e.g. amount washing liquid to use) **and** we want to provide the user with a GUI selector to set it.  Background: In the GUI and API the temp, spin and rinse select elements are not enabled until a course is selected that permits them to be overridden.  For example, if the course Cotton is selected temp, spin and rinse are enabled, while for the course Allergy Care only spin and rinse are enabled.  For each enabled override, a list of allowed values is extracted from from the course description.  When a user attempts to set a value it is checked against this list.  If the value is not in the list it is ignored and the user is informed of the permitted values by raising `ServiceValidationError`.

> I would prefer to just add some parameter to the service call and allow to change this value calling the HA service with prober parameters, that will be ignored if not valid

When I started this work you had not publish the change to `remote_start` that allowed a course name string to be passed as a parameter.  I have since studied your change and learnt how an integration can define a service that can be used by an automation or script.

I feel it is very easy to add another parameter to `remote_start` that accepts a JSON dictionary of overrides.  The method can check `_course_overrides_lists` to confirm if the setting (e.g. spin speed) can be overridden and that the value is correct.  An example call would be `remote_start('Eco 40-60', '{"temp": "TEMP_60", "spin": "SPIN_400"}')` 

If you are happy with the above approach I am happy to modify the PR.  I feel it will be a simple change. If you are thinking of something different please let me know.

I feel there is one problem we should agree on.  For course names users can discover the valid course names that can be passed to `remote_start` using the course selection entity on the integration GUI.  There may be another method that I have not found.  How will a user, for a given course, discover setting that can be overridden and their permitted values that can be passed to `remote_start`. The selectors for temp, spin and rinse in the integration GUI will provide this information to the user.  If you don't wish to have these, how do you think we can expose the information in `_course_overrides_lists` to the user?

In summary:
1. I like the idea of adding overrides to `remote_start`.  It gives a cleaner HA script.
2. I believe the code will handle all current and future course settings and WM models.
3. Without a discovery mechanism for the overrides that is simple for the user I don't feel we have usable solution. I am therefore reluctant to remove the temp, spin and rinse selectors if we don't provide an alternative. Nevertheless, this is your call.

Regards
John